smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelValidator.java

```diff
@@ -93,6 +97,29 @@ public ModelValidator addValidator(Validator validator) {
         return this;
     }
 
+    /**
+     * Adds a custom {@link ValidationEventDecorator} to the ModelValidator.
+     *
+     * @param decorator Decorator to add.
+     * @return Returns the ModelValidator.
+     */
+    public ModelValidator addValidationEventDecorator(ValidationEventDecorator decorator) {
```

Comment:
Since this class is package-private, you could probably get rid of this method or the other method and just keep the method that's actually used by ModelAssembler.

Reply:
I think we need to have both if we want to allow the user to configure directly in the assembler the decorators since we need to pass down to the `ModelValidator` the configured values + the loaded ones, or none if we are willing to go without this. I could go with none for the time being and add it if and when the need arises.

Reply:
I did remove methods to manually add decorators from the model assembler and validator for the time being, we can add those back if needed.